exercise/api/views.py

```diff
@@ -424,6 +428,39 @@ def grader_detail(self, request, *args, **kwargs):
 
         return Response(_post_async_submission(request, self.submission.exercise, self.submission))
 
+    @action(
+        detail=True,
+        url_path='re-submit',
+        url_name='re-submit',
+        get_permissions = lambda: [OnlyCourseStaffPermission()],
+        methods=['get'],
+    )
+    def resubmit(self, request, *args, **kwargs):
+        if not self.submission.exercise.is_submittable:
+            return self.http_method_not_allowed(request, *args, **kwargs)
+
+        data = None
+
+        page = self.submission.exercise.grade(self.submission, request)
+
+        # Enroll after succesfully resubmitting to the enrollment exercise.
+        if self.submission.exercise.status in (
+            LearningObject.STATUS.ENROLLMENT,
+            LearningObject.STATUS.ENROLLMENT_EXTERNAL,
+        ) and self.submission.status == Submission.STATUS.READY:
+            self.instance.enroll_student(request.user)
+
+        status_code = status.HTTP_200_OK
+        headers = {
+            'Location': reverse('api:submission-detail', kwargs={
```

Comment:
Why did you add the http redirect header to this API endpoint? How is it necessary here?

(I thought APIs usually don't redirect. They can return some success/failure message in the response to the client.)

Reply:
I added the header because I implemented the ``resubmit`` API endpoint the same way as the above ``submit`` endpoint:
https://github.com/apluslms/a-plus/blob/e832ba304b22392e8acc0e47292a4ccd3dae24b0/exercise/api/views.py#L264-L312

Is there a reason the redirect header is present in the ``submit`` endpoint, but shouldn't be included in ``resubmit``?